### DISCUSSION TITLE
The ability to update <Script /> component's src

### DISCUSSION THREAD

**Discussion by moaazbhnas228** [vercel/next.js#64612] [Ideas]
### Describe the feature you'd like to request

We're trying to integrate Google Analytics into our multi-language website. 
Each country has a different `id` that we use to upload its relevant script via Next's `<Script />` component.

``` js
import Script from "next/script";

function GoogleAnalytics({ id }) {
  return (
    <Script
      strategy="afterInteractive"
      src={`https://www.googletagmanager.com/gtag/js?id=${id}`}
    />
  );
}

export default GoogleAnalytics;

```

**The problem** is that when the id changes, the `<Script />`'s `src` stays with the first value it received.

### Describe the solution you'd like

Is it possible that when the `id` changes, the first script gets replaced by a new script with the new `src` value?

### Describe alternatives you've considered

We tried injecting native HTML `<script>` tag in `useEffect()`, but it didn't execute.

**Comment by Ansvyager** [comment]
Hi ! I really want to join this open project, how can i do to apply ?

**Comment by guitheengineer** [comment]
One alternative is using dangerouslySetInnerHTML:
https://codesandbox.io/p/sandbox/great-lena-b6j4mo?selection=%5B%7B%22endColumn%22%3A14%2C%22endLineNumber%22%3A24%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A16%7D%5D&file=%2Fpages%2Findex.tsx

```jsx
      <div
        dangerouslySetInnerHTML={{
          __html: `
          <script
            src="https://www.googletagmanager.com/gtag/js?id=${id}"
          ></script>
        `,
        }}
      ></div>
```

**Comment by moaazbhnas228** [comment]
@guilhermefront Would that remove the previous script?

**Comment by moaazbhnas228** [comment]
@Ansvyager What do you mean? If you want to contribute to Next.js, I recommend you start with those introductory issues: [issues/good-first-issues](https://github.com/vercel/next.js/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)

**Comment by guitheengineer** [comment]
> @guilhermefront Would that remove the previous script?

I tested here and it replaces that script. [See this codesandbox if you want to try anything.](https://codesandbox.io/p/sandbox/great-lena-b6j4mo?selection=%5B%7B%22endColumn%22%3A14%2C%22endLineNumber%22%3A24%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A16%7D%5D&file=%2Fpages%2Findex.tsx)

**Comment by moaazbhnas228** [comment]
@guilhermefront Thank you for your answer. It's the best we got so far, but unfortunately, when we tried it, the old script's variables and functions are still in the memory, so there has to be a reload. 
This is not Next.js related.
Thank you so much for your assistance
